client/apps/settings/client/curations/promoted_admin.jsx

```diff
@@ -0,0 +1,152 @@
+import React from "react"
+import _ from "underscore"
+import set from "lodash.set"
+import { Grid, Col, Row } from "react-styled-flexboxgrid"
+import DropdownHeader from "client/apps/edit/components/admin/components/dropdown_header.coffee"
+import ImageUpload from "client/apps/edit/components/admin/components/image_upload.coffee"
+import CharacterLimitInput from "client/components/character_limit/index.jsx"
+
+export default class PromotedAdmin extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      curation: props.curation,
+      saveStatus: "Save"
+    }
+  }
+
+  onImageInputChange(name, url, i) {
+    return this.onChange(name, url, i)
+  }
+
+  onInputChange(e, i) {
+    return this.onChange(e.target.name, e.target.value, i)
+  }
+
+  onChange(key, value, index) {
+    const newCuration = this.state.curation.clone()
+    const section = newCuration.get("sections")[index]
+    set(section, key, value)
+
+    return this.setState({
+      curation: newCuration,
+      saveStatus: "Save"
+    })
+  }
+
+  save() {
+    return this.state.curation.save({}, {
+      success: () => this.setState({ saveStatus: "Saved" }),
+      error: error => this.setState({ saveStatus: "Error" })
+    })
+  }
+
+  render() {
+    return (
+      <div>
+        {_.map(this.state.curation.get("sections"), (section, index) =>
+          <div className="promoted-admin__container admin-form-container" key={`promoted-admin-${index}`}>
+            <DropdownHeader
+              section={section.name}
+              key={`dropdown-${index}`}
+              className="promoted-admin__section-header" />
+            <div className="promoted-admin__section">
+              <Row key={index}>
+                <Col lg>
+                  <label>Title</label>
+                  <input
+                    className="bordered-input"
+                    placeholder="Title"
+                    defaultValue={section.name}
+                    onChange={ ((e) => this.onInputChange(e, index)).bind(this) }
+                    name="name" />
```

Comment:
Since the onChange events are repeated a few times each, it might make sense to move these into the constructor-- an [example here](https://github.com/eessex/positron/blob/9b458e2c779b2c8390915da3b27f50bba9a7b7f5/client/components/file_input/index.jsx#L8).

Reply:
Or even better, [omit constructors completely](https://github.com/eessex/positron/blob/master/.babelrc#L14).

Since we're using ES6 and fat-arrow functions bind to the outer scope automatically, you can do this instead: 

```js
class Home extends Component {
  state = {
    count: 0
  }

  handleClick = () => {
    const count = count + 1

    this.setState({
      count 
    })
  }

  render () {
    return (
      <div onClick={this.handleClick}>
        Click!
      </div>
    )
  }
}
```

No more need to use constructors at all except in cases where you need to set the state via props.